### DISCUSSION TITLE
Custom Naming Strategies for collections MongoDB

### DISCUSSION THREAD

**Discussion by lacerdanycolas** [micronaut-projects/micronaut-data#1933] [Q&A] [answered]
Hi, 

Is there any configurable way to create custom Naming Strategies for collections, and set in annotation _@MappedEntity_ in micronaut-data-mongodb? 

I'm creating a custom class that implements NamingStrategy interface, but the collection is created using the default strategy  anyway. My attempt below:

**Naming Custom Strategy**
```
class CamelCaseStrategy : NamingStrategy {

    @NonNull
    override fun mappedName(@NonNull name: String?): String? {
        return CaseUtils.toCamelCase(name, false) //utils from apache-commons-text 
    }
}
```

**My Entity**
```
@MappedEntity(namingStrategy = CamelCaseStrategy::class)
data class MyObject(){
...
}

```

**Comment by radovanradic** [comment]
@lacerdanycolas 

"Since Micronaut Data pre-computes the table and column name mappings at compilation time the specified [NamingStrategy](https://micronaut-projects.github.io/micronaut-data/latest/api/io/micronaut/data/model/naming/NamingStrategy.html) implementation must be on the annotation processor classpath (annotationProcessor scope for Java or kapt for Kotlin)."

[from](https://micronaut-projects.github.io/micronaut-data/latest/guide/#sqlNaming)

**Reply by lacerdanycolas** [reply]
Hi @radovanradic

I'm probably configuring something wrong. Is there any example or way to do this? I tried to directly add the tag into kapt scope, like code below, but the project doesn't build

```
 <annotationProcessorPaths>
         <annotationProcessor>myOrg.model.strategies.CamelCaseStrategy </annotationProcessor>
```

**Comment by radovanradic** [comment, marked as answer]
@lacerdanycolas 
Here is an simple example I created using Micronaut Launch: https://github.com/radovanradic/demo-mongo-ns
I have noticed you are using Kotlin so this is what should be added for annotation processor:
```
kapt("com.example:demo-mongo-ns")
```
One issue I had with this is that first needed to add my custom NamingStrategy, build project so libs are available for the annotation processor and then add this ^ to build.gradle 
There is probably better way but I am not very experienced with the build so this was workaround for me. You can see in debug logs when running test that Mongo is creating collection using custom naming strategy. 

HTH,
Radovan
